server/service/src/processors/transfer/shipment/test.rs

```diff
@@ -140,25 +153,81 @@ async fn invoice_transfers() {
+            tester.check_inbound_return_created(&ctx.connection);
+            ctx.processors_trigger.await_events_processed().await;
+            tester.check_outbound_return_was_linked(&ctx.connection);
+            tester.update_outbound_return_line(&service_provider);
+            tester.update_outbound_return_to_shipped(&service_provider);
+            ctx.processors_trigger.await_events_processed().await;
+            tester.check_inbound_return_was_updated(&ctx.connection);
+            tester.update_inbound_return_to_delivered(&service_provider);
+            ctx.processors_trigger.await_events_processed().await;
+            tester.check_outbound_return_status_matches_inbound_return(&ctx.connection);
+            tester.update_inbound_return_to_verified(&service_provider);
+            ctx.processors_trigger.await_events_processed().await;
+            tester.check_outbound_return_status_matches_inbound_return(&ctx.connection);
+
+            // With delete -- SHIPMENT
             let mut tester = ShipmentTransferTester::new(
                 &outbound_store,
                 &inbound_store,
+                Some(&outbound_store_name),
                 Some(&inbound_store_name),
                 &item1,
                 &item2,
             );
 
+            // Setup: create requisition
             tester.insert_request_requisition(&service_provider).await;
             ctx.processors_trigger.await_events_processed().await;
             tester.check_response_requisition_created(&ctx.connection);
+
+            // Create shipment, check it transfers, delete it, check inbound is deleted
             tester.insert_outbound_shipment(&ctx.connection);
             tester.update_outbound_shipment_to_picked(&service_provider);
             ctx.processors_trigger.await_events_processed().await;
             tester.check_inbound_shipment_created(&ctx.connection);
             tester.delete_outbound_shipment(&service_provider);
             ctx.processors_trigger.await_events_processed().await;
             tester.check_inbound_shipment_deleted(&ctx.connection);
+
+            // With delete -- RETURN
+            let mut tester = ShipmentTransferTester::new(
+                &outbound_store,
+                &inbound_store,
+                Some(&outbound_store_name),
+                Some(&inbound_store_name),
+                &item1,
+                &item2,
+            );
+            // Setup: create shipment
+            tester.insert_request_requisition(&service_provider).await;
+            ctx.processors_trigger.await_events_processed().await;
+            tester.check_response_requisition_created(&ctx.connection);
+            tester.insert_outbound_shipment(&ctx.connection);
+            tester.update_outbound_shipment_to_picked(&service_provider);
+            ctx.processors_trigger.await_events_processed().await;
+            tester.check_inbound_shipment_created(&ctx.connection);
```

Comment:
This actually raised a bit of a flag - we can create an Outbound Return from an Inbound Shipment that's actually only in `Picked` status (I would assume you could create one from a manual IS in `New` status)

The UI prevents this from happening, but should the backend prevent someone returning lines from a shipment before its verified?